src/include/gurt/debug_setup.h

```diff
@@ -0,0 +1,270 @@
+ *
+ * 5. Neither the name of Intel Corporation, nor the name of any Contributor
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef __GURT_DEBUG_SETUP_H__
+#define __GURT_DEBUG_SETUP_H__
+
+#if defined(D_LOG_V2) && !defined(D_LOG_V1_TEST)
+#define D_LOG_USE_V2
+#endif
+
+
+/**
+ * \file
+ * Debug setup macros
+ */
+
+/** @addtogroup GURT_DEBUG
+ * @{
+ */
+#define DD_FAC(name)	d_##name##_logfac
+
+#define DD_FAC_DECL(name)	DD_FAC(name)
+
+#ifndef D_LOGFAC
+#define D_LOGFAC	DD_FAC(misc)
+#endif
+
+#define DD_CONCAT_CACHE(x, y) x ##_cache
+#define DD_CONCAT_FLAG(x, y) DD_FLAG_## x ##_## y
+#define DD_CONCAT(x, y, op) op(x, y)
+#define DD_CACHE(fac)	\
+	DD_CONCAT(fac, D_NOOP, DD_CONCAT_CACHE)
+#define DD_FLAG_NAME(mask, fac)	\
+	DD_CONCAT(mask, fac, DD_CONCAT_FLAG)
+#define DD_FLAG(mask, fac)	\
+	DD_CACHE(fac)[DD_FLAG_NAME(mask, fac) - 1]
+
+/* These macros are intended to be used with FOREACH macros that define
+ * debug bits for your library.   See D_FOREACH_DB for an example.  Not
+ * all fields are used in individual macros.
+ *
+ * These macros can also be used standalone but are not quite as convenient
+ * due to required but unused arguments
+ */
+
+/** Declare an extern for a debug flag variable
+ *
+ * \param flag		The name of the variable used to store the value of the
+ *			debug bit.
+ * \param s_name	Unused
+ * \param l_name	Unused
+ * \param mask		Unused
+ * \param arg		Unused
+ */
+#define DD_DECLARE_DB(flag, s_name, l_name, mask, arg)	\
+	extern d_dbug_t	flag;
+
+/** Instantiate a debug flag variable
+ *
+ * \param flag		The name of the variable used to store the value of the
+ *			debug bit.
+ * \param s_name	Unused
+ * \param l_name	Unused
+ * \param mask		Unused
+ * \param arg		Unused
+ */
+#define DD_INSTANTIATE_DB(flag, s_name, l_name, mask, arg)	\
+	d_dbug_t	flag;
+
+/** Internal use macro as part of DD_REGISTER_DB.  It's multiline on purpose
+ *  so we can break from the foreach loop.  Checkpatch will complain but don't
+ *  see a better way.
+ */
+#define ALLOCATE_DBG_BIT(dbgbit, s_name, l_name, mask, rc)		\
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop